library/choose_disk.py

```diff
@@ -0,0 +1,651 @@
+        if device_name in physical_disks:
+            if device_name not in matching_devices:
+                matching_devices[device_name] = [f]
+            else:
+                matching_devices[device_name].append(f)
+
+    # Saving matching disks and add the path of the block device
+    # Output is like : Renaming        vdb to                        virtio-e0b94c7d-75ca-4175-a
+    for physical_disk in sorted(physical_disks):
+        if physical_disk in matching_devices:
+            current_index = sorted(matching_devices[physical_disk])[0]
+            final_disks[current_index] = physical_disks[physical_disk]
+            final_disks[current_index]["bdev"] = "%s%s" % (directory, current_index)
+            logger.info(' Renaming %10s to %50s', physical_disk, current_index)
+        else:
+            current_index = physical_disk
+            final_disks[current_index] = physical_disks[physical_disk]
+
+    return final_disks
+
+
+def fake_device(legacy_devices, ceph_type):
+    '''
+    In case of legacy block device names, let's create an internal faked
+    entry with a 'bdev' entry filled with the actual path. This will be used to
+    make a match later on.
+    '''
+    devices = {}
+    count = 0
+    for device in legacy_devices:
+        devices["%s_%d" % (ceph_type, count)] = {"bdev": os.path.dirname(device)+"/"+os.path.basename(device)}
+        count = count + 1
+
+    return devices
+
+
+def show_resulting_devices(matched_devices, physical_disks):
+    '''
+    Print the current state :
+      - what devices matched
+      - what devices didn't matched
+    '''
+    bdev_matched = []
+    bdev_unmatched = []
+
+    def prepare_device_string(device):
+        device_string = ""
+        if "ceph_prepared" in device:
+            device_string += " (ceph %s)" % device["ceph_prepared"]
+        device_string += " : "
+        for feature in ["vendor", "model", "size", "rotational"]:
+            if feature in device:
+                device_string += "%s:%s " % (feature[0], device[feature])
+
+        return device_string
+
+    logger.info("Matched devices   : %3d", len(matched_devices))
+    for matched_device in sorted(matched_devices):
+        logger.info(" %s : %s%s", matched_device, matched_devices[matched_device]["bdev"], prepare_device_string(matched_devices[matched_device]))
+        bdev_matched.append(matched_devices[matched_device]["bdev"])
+
+    for physical_disk in sorted(physical_disks):
+        pdisk = physical_disks[physical_disk]
+        if pdisk["bdev"] not in bdev_matched:
+            bdev_unmatched.append("%s %s" % (pdisk["bdev"], prepare_device_string(pdisk)))
+
+    logger.info("Unmatched devices : %3d", len(bdev_unmatched))
+    for bdev in sorted(bdev_unmatched):
+        logger.info(" %s", bdev)
+
+
+def setup_logging():
+    '''
+    Preparing the logging system
+    '''
+    hdlr = logging.FileHandler('/var/log/choose_disk.log')
```

Comment:
this will require that current user to have permissions in `/var/log/choose_disk.log` or `/var/log`. If that hasn't been set before it will imply the need for super user privileges, while the rest of ceph-ansible doesn't require it. This should be optional, with a default that doesn't force a user to run with super user privileges